### ISSUE TITLE
MA0053: Why are records not included in the check?

### ISSUE THREAD

**Issue by Code-Chops** [meziantou/Meziantou.Analyzer#914]
### Version of the Meziantou.Analyzer NuGet package

2.0.220

### Rule Identifier

MA0053

### Target Framework

.NET 7

### C# Language version

C# 13

### Description

Only classes are being checked for having the`sealed` keyword. Why are records not included?

Thank you by the way for the great analyser!

**Comment by meziantou** [comment]
The rule should already support classes and records. Please provide a small repro if you don't get a diagnostic.

That's being said, the documentation should be improved.

**Comment by Code-Chops** [comment]
```
// Does not give a warning
public record Record;
```

```
// Gives a warning
public class Class;
```

**Comment by meziantou** [comment]
Indeed, it doesn't work. I'm not sure why as there is a test to validate the behavior. Maybe something changed in recent version of Roslyn.

I think the issue is that a record contains virtual members, which prevent the analyzer from reporting it. You can try adding the following configuration:

```
MA0053.class_with_virtual_member_should_be_sealed = true
```

**Comment by meziantou** [comment]
The latest version should better handle records: https://github.com/meziantou/Meziantou.Analyzer/pull/915

**Comment by Code-Chops** [comment]
Thank you for fixing this!
